op_crates/fetch/23_request.js

```diff
@@ -0,0 +1,521 @@
+        if (
+          temporaryMimeType === null ||
+          mimesniff.essence(temporaryMimeType) == "*/*"
+        ) {
+          continue;
+        }
+        mimeType = temporaryMimeType;
+        if (mimesniff.essence(mimeType) !== essence) {
+          charset = null;
+          const newCharset = mimeType.parameters.get("charset");
+          if (newCharset !== undefined) {
+            charset = newCharset;
+          }
+          essence = mimesniff.essence(mimeType);
+        } else {
+          if (mimeType.parameters.has("charset") === null && charset !== null) {
+            mimeType.parameters.set("charset", charset);
+          }
+        }
+      }
+      if (mimeType === null) return null;
+      return mimeType;
+    }
+    get [_body]() {
+      return this[_request].body;
+    }
+
+    /**
+     * https://fetch.spec.whatwg.org/#dom-request
+     * @param {RequestInfo} input 
+     * @param {RequestInit} init 
+     */
+    constructor(input, init = {}) {
+      const prefix = "Failed to construct 'Request'";
+      webidl.requiredArguments(arguments.length, 1, { prefix });
+      input = webidl.converters["RequestInfo"](input, {
+        prefix,
+        context: "Argument 1",
+      });
+      init = webidl.converters["RequestInit"](init, {
+        prefix,
+        context: "Argument 2",
+      });
+
+      this[webidl.brand] = webidl.brand;
+
+      /** @type {InnerRequest} */
+      let request;
+      const baseURL = getLocationHref();
+
+      // 5.
+      if (typeof input === "string") {
+        const parsedURL = new URL(input, baseURL);
+        request = newInnerRequest("GET", parsedURL.href, [], null);
+      } else { // 6.
+        if (!(input instanceof Request)) throw new TypeError("Unreachable");
+        request = input[_request];
+      }
+
+      // 22.
+      if (init.redirect !== undefined) {
+        request.redirectMode = init.redirect;
+      }
+
+      // 25.
+      if (init.method !== undefined) {
+        let method = init.method;
+        method = validateAndNormalizeMethod(method);
+        request.method = method;
+      }
+
+      // NOTE: non standard extension. This handles Deno.HttpClient parameter
+      if (init.client !== undefined) {
+        if (init.client !== null && !(init.client instanceof HttpClient)) {
+          throw webidl.makeException(
+            TypeError,
+            "`client` must be a Deno.HttpClient",
+            { prefix, context: "Argument 2" },
+          );
+        }
+        request.clientRid = init.client?.rid ?? null;
+      }
+
+      // 27.
+      this[_request] = request;
+
+      // 29.
+      this[_headers] = headersFromHeaderList(request.headerList, "request");
+
+      // 31.
+      if (Object.keys(init).length > 0) {
+        let headers = headerListFromHeaders(this[_headers]);
+        if (init.headers !== undefined) {
+          headers = init.headers;
+        }
+        headerListFromHeaders(this[_headers]).slice(
```

Comment:
I can't tell which .slice method this is, but does it mutate the source? If not, this is dead code.